### DISCUSSION TITLE
Build fails: no such file or directory 'index.html'

### DISCUSSION THREAD

**Discussion by JKarlavige** [vercel/community#3214] [Help] [answered]
### Summary

I am transferring a Netlify site to Vercel. During the Vercel build, I am receiving the error: `Error: ENOENT: no such file or directory, open '/vercel/path0/.next/server/pages/index.html'`.

In our NextJS project we use a page builder to build pages dynamically, including our homepage. So an `index.html` does not exist within our `pages` directory. The page ultimately comes from our `pages/[...slug].tsx` dynamic NextJS file. 

If I build the site locally with `npm run build`, I see the index page in the build folder at `.next/server/pages/index/index.html`.

It appears the Vercel build is expecting the `index.html` page to be under `pages`, however the NextJS build is putting the index page in its own directory.

I am using all default build settings for the Vercel project. Any help would be appreciated!

Thank you,
Jason 

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/13647110/4ba7404f-3e82-4ba0-b6c4-f9d185307d86)

**Comment by amyegan** [comment]
Hi @JKarlavige! It sounds like you're using a page builder tool that I may not be familiar with. If you can share a [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) in a public repo, that will give me a better understanding of the project setup and how everything should work together.

**Reply by JKarlavige** [reply]
Hi @amyegan, I was able to recreate the error in an [example repo](https://github.com/JKarlavige/nextjs-debug-test) without the CMS. 

There are two files I setup to mimic our production setup:
- [pages/[...slug].tsx](https://github.com/JKarlavige/nextjs-debug-test/blob/main/pages/%5B...slug%5D.tsx)
- [next.config.js](https://github.com/JKarlavige/nextjs-debug-test/blob/main/next.config.js)

In the `pages/[...slug].tsx` file under `getStaticPaths`, we are building the paths based on a `slug` field in the CMS. For the homepage, we didn't find a way to build the homepage dynamically with an empty path or by using `/`. So we're setting the slug as `index` for the homepage.

Within `next.config.js`, we are rewriting any requests made to `/` over to `/index` where our homepage is built. Then, to prevent users from landing on `/index`, we're redirecting all requests to `/`. This serves our homepage at `https://localhost:3000`, and any requests made to `https://localhost:3000/index` are redirected back to `https://localhost:3000`.

With this setup, after running `npm run build`, the index page is set at `.next/server/pages/index/index.html`. During the Vercel build, I receive the error: `Error: ENOENT: no such file or directory, open '/vercel/path0/.next/server/pages/index.html'`.

This only appears to happen on Vercel, as we have been using this setup on Netlify in production for the past few months. _I'm also not convinced the rewrite/redirect setup is optimal, and open to suggestions on improving how we can dynamically build the homepage._

Let me know if i can help clarify anything else! Thanks again for your help.

**Reply by JKarlavige** [reply, marked as answer]
Hey @amyegan, update on this. I came across [Optional Catch-all routes](https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes#optional-catch-all-segments) which use two brackets rather than one.

After updating our page to `[[...slug]].tsx` and adjusting our `getStaticPaths` to prevent building an `index` page from the slug, our build began placing the `index.html` directly in the pages directory and the Vercel build now completes successfully.

This also allows us to remove the skeptical rewrite we had setup.

Thank you again for your help!
